desidlas/training/training.py

```diff
@@ -0,0 +1,397 @@
+                    best_offset_rmse = result_rmse_offset if result_rmse_offset < best_offset_rmse else best_offset_rmse
+                    best_density_rmse = result_rmse_coldensity if result_rmse_coldensity < best_density_rmse else best_density_rmse
+                    
+                    print("             test accuracy/offset RMSE/density RMSE:     {:0.3f} / {:0.3f} / {:0.3f}\n".format(
+                          test_accuracy, result_rmse_offset, result_rmse_coldensity))
+                    record_file.write("             test accuracy/offset RMSE/density RMSE:     {:0.3f} / {:0.3f} / {:0.3f}".format(
+                          test_accuracy, result_rmse_offset, result_rmse_coldensity))
+                    
+                    save_checkpoint(sess, (save_filename + "_" + str(i)) if save_filename is not None else None)
+
+           # Save checkpoint
+            save_checkpoint(sess, save_filename)
+
+            return best_accuracy, test_accuracy, np.mean(loss_value), best_offset_rmse, result_rmse_offset, \
+                   best_density_rmse, result_rmse_coldensity
+
+def save_checkpoint(sess, save_filename):
+    """
+    save model file
+
+    Parameters
+    ----------
+    sess:tf.Session object
+    save_filename:the path to save the model
+
+    Returns
+    -------
+
+    """
+    if save_filename is not None:
+        tf.compat.v1.train.Saver().save(sess, save_filename + ".ckpt")
+        with open(checkpoint_filename + "_hyperparams.json", 'w') as fp:
+            json.dump(hyperparameters, fp)
+        print("Model saved in file: %s" % save_filename + ".ckpt")
+
+def calc_normalized_score(best_accuracy, best_offset_rmse, best_coldensity_rmse):
+    """
+    Generate the normalized score from the 3 loss functions
+
+    This should be used for a hyperparameter search
+
+    Parameters
+    ----------
+    best_accuracy:the highest classification accuracy during the training
+    best_offset_rmse:the minimum offset RMSE during the training
+    best_coldensity_rmse:the minimum column density during the training
+
+    Returns:
+    -------
+    normalized_score:used to select hyperparameters. A better hyperparameter combine will have a smaller normalized_score
+
+    """
+    # These mean & std dev are used to normalize the score from all 3 loss functions for hyperparam optimize
+    # TODO -- Reconsider these for DESI
+    mean_best_accuracy = 0.02
+    std_best_accuracy = 0.0535
+    mean_best_offset_rmse = 4.0
+    std_best_offset_rmse = 1.56
+    mean_best_coldensity_rmse = 0.33
+    std_best_coldensity_rmse = 0.1
+    normalized_score = (1 - best_accuracy - mean_best_accuracy) / std_best_accuracy + \
+                       (best_offset_rmse - mean_best_offset_rmse) / std_best_offset_rmse + \
+                       (best_coldensity_rmse - mean_best_coldensity_rmse) / std_best_coldensity_rmse
+    
+    return normalized_score
+
+
+
+if __name__ == '__main__':
```

Comment:
The code that follows contains a lot of hard coded info, e.g. paths.

Please make this a separate script.

Reply:
Thanks. I have replaced the hard coded info.